G/Git/build_tarballs.jl

```diff
@@ -43,24 +44,49 @@ else
     sed -i 's/cross_compiling=yes/cross_compiling=no/' configure
 fi
 
-if [[ "${target}" == *-apple-* ]]; then
-    LDFLAGS="-lgettextlib"
-elif [[ ${target} == *-freebsd* ]]; then
-    LDFLAGS="-L${prefix}/lib -lcharset"
-fi
-
 ./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} \
     --with-curl \
     --with-expat \
     --with-openssl \
     --with-iconv=${prefix} \
     --with-libpcre2 \
     --with-zlib=${prefix} \
-    "${CACHE_VALUES[@]}" \
-    CPPFLAGS="-I${prefix}/include" \
-    LDFLAGS="${LDFLAGS}"
+    "${CACHE_VALUES[@]}"
 make -j${nproc}
 make install INSTALL_SYMLINKS="yes, please"
+
+# Because of the System Integrity Protection (SIP), when running shell or Perl scripts, the
+# environment variable `DYLD_FALLBACK_LIBRARY_PATH` is reset.  We work around this
+# limitation by re-exporting `DYLD_FALLBACK_LIBRARY_PATH` through another environment
+# variable called `JLL_DYLD_FALLBACK_LIBRARY_PATH` which won't be reset by SIP.
+# Read more about the issue and the hacks we apply here at
+# <https://github.com/JuliaVersionControl/Git.jl/issues/40>.
+if [[ "${target}" == *-apple-* ]]; then
+    for file in $(grep -l -r '^#!/bin/sh' "${prefix}"); do
+        if [[ $(basename "${file}") == git* ]]; then
+            sed -i -e 's?^#!/bin/sh?#!/bin/sh\nexport DYLD_FALLBACK_LIBRARY_PATH="${JLL_DYLD_FALLBACK_LIBRARY_PATH}"?' "${file}"
+        fi
+    done
+    for file in $(grep -l -r '^#!/usr/bin/perl' "${prefix}"); do
+        if [[ $(basename "${file}") == git* ]]; then
+            sed -i -e 's?^#!/usr/bin/perl?#!/usr/bin/perl\n$ENV{'DYLD_FALLBACK_LIBRARY_PATH'} = $ENV{'JLL_DYLD_FALLBACK_LIBRARY_PATH'};?' "${file}"
+        fi
+    done
```

Comment:
I think with the `git` renaming, you don't need these lines anymore

Reply:
Ok, yes, I can confirm it works also without these lines.